packages/examples/cvat/exchange-oracle/src/handlers/escrow_cleanup.py

```diff
@@ -0,0 +1,63 @@
+from __future__ import annotations
+
+import contextlib
+from collections.abc import Callable
+from dataclasses import dataclass
+from typing import TYPE_CHECKING
+
+from cvat_sdk.api_client.exceptions import NotFoundException
+
+import src.cvat.api_calls as cvat_api
+import src.services.cloud as cloud_service
+from src.core.config import Config
+from src.core.storage import (
+    compose_data_bucket_prefix,
+    compose_results_bucket_prefix,
+)
+from src.services.cloud.utils import BucketAccessInfo
+
+if TYPE_CHECKING:
+    from src.models.cvat import Project
+
+
+@dataclass
+class EscrowCleaner:
+    escrow_address: str
+    chain_id: int
+    projects: list[Project]
+
+    def cleanup(self) -> None:
+        self._cleanup_cvat()
+        self._cleanup_storage()
```

Comment:
I believe we should perform all the steps regardless of errors on any of the steps. Maybe, re-raise the error later, if anything happened. Otherwise we can end up with a half-cleaned escrow.

Reply:
The reasoning behind this way of handling things was to make sure that everything that might use these files (for instance CVAT), would be cleared first, so there would not be a condition when files are still referenced and used somewhere after they've been deleted from the bucket. 

If that's not a problem, of course we can clean them regardless. Any thoughts on this?

Reply:
No, I don't think it's a problem. There should be no access to the project after it's marked deleted. It can be considered an error, if there is any access happening, it's not expected to work.